simp_le.py

```diff
@@ -78,6 +82,26 @@ class TestCase(unittest.TestCase):
     """simp_le unit test case."""
 
 
+@contextlib.contextmanager
+def temp_umask(umask):
+    """Context manager that temporarily sets the process umask."""
+    oldmask = os.umask(umask)
+    try:
+        yield
+    finally:
+        os.umask(oldmask)
+
+
+def open_keyfile(filename, mode):
+    """Opens a sensitive file for writing."""
+    flags = os.O_WRONLY | os.O_CREAT | os.O_TRUNC
+    # Windows-only
+    if 'b' in mode and hasattr(os, "O_BINARY"):
+        flags |= os.O_BINARY  # pylint: disable=no-member
+    fild = os.open(filename, flags, 0o600)
```

Comment:
inlining, ok.

600 should be the default, at least postgres (or was it just the node-pg or PHP library?) require such strict permissions. `ssh-keygen` also creates such files. And for simplicity reasons, I would not add another option to change it.

Rationale: you can chmod afterwards, or create an empty file before that with appropriate permissions. (Actually, empty files seem to trigger `load_from_content("")`, maybe that should also return `EMPTY_DATA`?)